components/suggest/src/suggestion.rs

```diff
@@ -96,13 +96,35 @@ pub enum Suggestion {
         icon: Option<Vec<u8>>,
         icon_mimetype: Option<String>,
         score: f64,
+        match_info: FtsMatchInfo,
     },
     Exposure {
         suggestion_type: String,
         score: f64,
     },
 }
 
+/// Additional data about how an FTS match was made(https://bugzilla.mozilla.org/show_bug.cgi?id=1931373)
+#[derive(Debug, Clone, PartialEq, uniffi::Record)]
+pub struct FtsMatchInfo {
+    /// Was this a prefix match (`water b` matched against `water bottle`)
+    pub prefix: bool,
+    /// Did the match require stemming? (`run shows` matched against `running shoes`)
+    pub stemming: bool,
+    /// How far apart were the terms product title
+    pub term_distance: FtsTermDistance,
+}
+
+#[derive(Debug, Clone, PartialEq, uniffi::Enum)]
+pub enum FtsTermDistance {
+    // All terms in a 3-term chunk
+    Near,
+    // All terms in a 5-term chunk
+    Medium,
+    // No 5-term chunk that contains all the terms
+    Far,
```

Comment:
The segments above were arbitrarily picked by me.  Are there different numbers we should choose?

Reply:
I don't know, what's the intuition behind collecting this distance data?

3 seems kind of like a large distance to start with. If I'm reading the fts5 doc right, that means terms can have 3 words between them and the match will still be successful. I would think we'd want to start at 1 at most, maybe even 0?

Is there a reason for using an enum and not reporting the numeric distance itself, as like `min_term_distance`?

How hard would it be to add tests for non-near distances?

Reply:
Good catch on this one, the docs confused me and I thought `3` meant the total amount of words in the clump was 3.  I changed these numbers to `1` and `3`, which seems like more of a reasonable start.

That said, I'm still not sure what's correct to test for, I'm open to changing "Near" to meaning `0`.

The one thing I don't think we can do is calculate actual minimum term distance number.  AFAICT, there's no function for that, you just have to make a bunch of queries and see if they match or not.  Maybe we have variants like `Adjacent=0`, `Medium=1` and `Far=2 or greater` though.